addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURL.java

```diff
@@ -200,19 +215,21 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                 }
             }
         }
+
         if (!found && msg.getRequestHeader().getURI().getEscapedPath() != null) {
             // Handle jsessionid like:
             // http://tld.gtld/fred;jsessionid=1A530637289A03B07199A44E8D531427?foo=bar
             Matcher jsessMatcher = null;
+            String path = "";
             try {
-                jsessMatcher =
-                        Pattern.compile("jsessionid=[\\dA-Z]*", Pattern.CASE_INSENSITIVE)
-                                .matcher(msg.getRequestHeader().getURI().getPath());
+                // We lowercase the path because the regex has a specific session ID
+                // in it which can't be made case insensitive
+                path = msg.getRequestHeader().getURI().getPath().toLowerCase();
+                jsessMatcher = PATHSESSIONIDPATTERN.matcher(path);
             } catch (URIException e) {
+                e.printStackTrace();
             }
-            if (jsessMatcher != null
-                    && jsessMatcher.find()
-                    && sessionIds.contains(jsessMatcher.group().split("=")[0].trim())) {
```

Comment:
This validation was removed.

Reply:
Can you explain what the sessionIds.contains() check that was removed is supposed to do? Its not obvious from the code and seemed pointless, so we removed it. Happy to put it back if you think it actually does something useful/important.

Reply:
It validates that the `jsessionid` is one of the declared session token names, allowing the user to opt-out this check when removed/disabled (like in the previous check we only consider the declared session token names). I guess we can argue that it's better to alert always in this specific case.

Reply:
Worth adding a test for that case (i.e. the alert is still raised even if `jsessionid` is disabled/removed).

Reply:
OK. I see how its getting the options in the scanner:
        OptionsParam options = Model.getSingleton().getOptionsParam();
        HttpSessionsParam sessionOptions = options.getParamSet(HttpSessionsParam.class);

But in the unit test, how would I set this parameter set to be either null, or empty, or have a set of session IDs that does not contain JSessionID?

Reply:
You can remove all with `sessionOptions.setDefaultTokens(Collections.emptyList());`.

Reply:
When I do this, with this code:

        OptionsParam options = Model.getSingleton().getOptionsParam();
        HttpSessionsParam sessionOptions = options.getParamSet(HttpSessionsParam.class);
        sessionOptions.setDefaultTokens(Collections.emptyList()); <-- this is: TestInfoSessionIdURLUnitTest.java:220

I get this error:

org.zaproxy.zap.extension.pscanrules.TestInfoSessionIdURLUnitTest > alertsWithoutJSessionidInOptions FAILED
    java.lang.NullPointerException
        at org.zaproxy.zap.extension.httpsessions.HttpSessionsParam.setDefaultTokens(HttpSessionsParam.java:147)
        at org.zaproxy.zap.extension.pscanrules.TestInfoSessionIdURLUnitTest.alertsWithoutJSessionidInOptions(TestInfoSessionIdURLUnitTest.java:220)

Are we doing it wrong or is there a bug in the setDefaultTokens() method of: HttpSessionsParam?

Reply:
Sorry, should have mentioned that, the `HttpSessionsParam` needs to be added after loading the configuration, in `setUpHttpSessionsParam` it should do:
```Java
OptionsParam options = Model.getSingleton().getOptionsParam();
options.load(new ZapXmlConfiguration());
options.addParamSet(new HttpSessionsParam());
```